src/api.ts

```diff
@@ -167,9 +168,40 @@ class RippleAPI extends EventEmitter {
     })
   }
 
+  /**
+   * Returns objects owned by an account.
+   * For an account's trust lines and balances,
+   * see `getTrustlines` and `getBalances`.
+   */
+  async requestAll(
+    command: 'account_objects',
+    params: AccountObjectsRequest
+  ): Promise<AccountObjectsResponse[]> {
+    // TODO: Prevent access to non-validated ledger versions
+    const results = await (<Function>this._requestAll)(command, params, {
+      collect: 'account_objects'
+    })
+
+    return results.reduce((result, singleResult) => {
+      const invariants = ['account', 'ledger_hash',
+        'ledger_index', 'ledger_current_index', 'validated']
+
+      invariants.forEach(field => {
+        if (result[field] !== singleResult[field]) {
+          throw new errors.UnexpectedError(
```

Comment:
Just being extra defensive. Since it should never happen (e.g. could be a faulty or malicious rippled server), it's important to fail early.

This is probably a moot point though since I think we should actually move to expose pagination in the ripple-lib API (https://github.com/ripple/ripple-lib/pull/861#issuecomment-370922847)